Content.Server/Explosion/EntitySystems/ExplosionSystem.cs

```diff
@@ -1,392 +1,289 @@
+    {
+        var spaceTiles = spaceData?.TileLists;
+
+        Dictionary<GridId, Dictionary<int, List<Vector2i>>> tileLists = new();
+        foreach (var grid in gridData)
         {
-            if (!Resolve(entity, ref transform))
-            {
-                return;
-            }
-
-            Resolve(entity, ref explosive, false);
-
-            if (explosive is { Exploding: false })
-            {
-                _triggers.Explode(entity, explosive, user);
-            }
-            else
-            {
-                while (EntityManager.EntityExists(entity) && entity.TryGetContainer(out var container))
-                {
-                    entity = container.Owner;
-                }
-
-                if (!EntityManager.TryGetComponent(entity, out transform))
-                {
-                    return;
-                }
-
-                var epicenter = transform.Coordinates;
-
-                SpawnExplosion(epicenter, devastationRange, heavyImpactRange, lightImpactRange, flashRange, entity, user);
-            }
+            tileLists.Add(grid.Grid.Index, grid.TileLists);
         }
 
-        public void SpawnExplosion(
-            EntityCoordinates epicenter,
-            int devastationRange = 0,
-            int heavyImpactRange = 0,
-            int lightImpactRange = 0,
-            int flashRange = 0,
-            EntityUid? entity = null,
-            EntityUid? user = null)
+        return new ExplosionEvent(_explosionCounter, epicenter, id, iterationIntensity, spaceTiles, tileLists, spaceMatrix, spaceData?.TileSize ?? DefaultTileSize);
+    }
+
+    private void CameraShake(float range, MapCoordinates epicenter, float totalIntensity)
+    {
+        var players = Filter.Empty();
+        players.AddInRange(epicenter, range, _playerManager, EntityManager);
+
+        foreach (var player in players.Recipients)
         {
-            var mapId = epicenter.GetMapId(EntityManager);
-            if (mapId == MapId.Nullspace)
-            {
-                return;
-            }
-
-            // logging
-            var range = $"{devastationRange}/{heavyImpactRange}/{lightImpactRange}/{flashRange}";
-            if (entity == null || !entity.Value.IsValid())
-            {
-                _logSystem.Add(LogType.Explosion, LogImpact.High, $"Explosion spawned at {epicenter:coordinates} with range {range}");
-            }
-            else if (user == null || !user.Value.IsValid())
-            {
-                _logSystem.Add(LogType.Explosion, LogImpact.High,
-                    $"{ToPrettyString(entity.Value):entity} exploded at {epicenter:coordinates} with range {range}");
-            }
-            else
-            {
-                _logSystem.Add(LogType.Explosion, LogImpact.High,
-                    $"{ToPrettyString(user.Value):user} caused {ToPrettyString(entity.Value):entity} to explode at {epicenter:coordinates} with range {range}");
-            }
```

Comment:
these appear to have been removed entirely? these ones are pretty important logs

Reply:
oh yeah completely forgot to re-add logs. will fix.